frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts

```diff
@@ -0,0 +1,108 @@
+  execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND}`).toString(),
+);
+
+let clusterJSON;
+let previousCnt;
+let previousPods;
+let updatedClusterJSON;
+let updatedCnt;
+let updatedPods;
+
+describe('Check add capacity functionality for ocs service', () => {
+  beforeAll(async () => {
+    // check whether the cluster is present or not, and check for its ready state
+    // if not present exit and console that no cluster is present, will be send in next PR
+
+    const { name } = storageCluster.items[0].metadata;
+    clusterJSON = JSON.parse(
+      execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND} ${name}`).toString(),
+    );
+    previousCnt = _.get(clusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+    const uid = _.get(clusterJSON, 'metadata.uid', undefined).toString();
+    previousPods = JSON.parse(execSync(`kubectl get pods -n ${NAMESPACE} -o json`).toString());
+
+    await browser.get(
+      `${appHost}/k8s/ns/openshift-storage/operators.coreos.com~v1alpha1~ClusterServiceVersion`,
+    );
+
+    await click(ocsOp);
+    await click(storageClusterView);
+
+    await browser.wait(until.presenceOf(storageClusterRow(uid)));
+    const kebabMenu = storageClusterRow(uid).$('button[data-test-id="kebab-button"]');
+
+    await click(kebabMenu);
+    await click(addCapacityLbl);
+    await click(addCapacityBtn);
+
+    updatedClusterJSON = JSON.parse(
+      execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND} ${name}`).toString(),
+    );
+
+    updatedCnt = _.get(updatedClusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+    const statusCol = storageClusterRow(uid).$('td:nth-child(4)');
+
+    // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
+    await browser.sleep(2000);
+
+    await browser.wait(until.textToBePresentInElement(statusCol, 'Progressing'));
+    await browser.wait(
+      until.textToBePresentInElement(statusCol.$('span.co-icon-and-text span'), 'Ready'),
+    );
+
+    updatedPods = JSON.parse(execSync(`kubectl get pod -o json -n ${NAMESPACE}`).toString());
+  }, EXPANDWAIT);
+
+  it('Newly added capacity should takes into effect at the storage level', () => {
+    // by default 2Tib capacity is being added
+    expect(updatedCnt - previousCnt).toEqual(1);
+  });
+
+  it('New osd pods corresponding to the additional capacity should be in running state', () => {
+    const isPodPresent = (podName) =>
+      previousPods.items.find((pod) => pod.metadata.name === podName) !== undefined;
+
+    const newPods = [];
+    updatedPods.items.forEach((pod) => {
+      if (!isPodPresent(pod.metadata.name) && pod.metadata.name.startsWith('rook-ceph-osd')) {
+        newPods.push(pod.metadata.name);
+      }
+    });
+
+    /* since rook-ceph-osd-prepare-ocs-deviceset- keeps changing their last 4 characters,
+    hence subtracting the count of previous  rook-ceph-osd-prepare-ocs-deviceset- pods */
+    expect(newPods.length - previousCnt * 3).toEqual(6);
```

Comment:
Doesn't look right to me. Can you elaborate how this works? Maybe add a comment to the code?

Reply:
For each iteration, the last 4 characters of the prepare pods name gets change. I have already added comment.

Reply:
How does this equation work is what I wanted to understand.

Reply:
Attaching screenshots for better clarity -
Before triggering next expansion
![before-expansion](https://user-images.githubusercontent.com/5517376/70145699-0de9c480-16e4-11ea-9e65-a24d44027fd4.png)

After expansion -
![after-expansion](https://user-images.githubusercontent.com/5517376/70145722-1510d280-16e4-11ea-8b17-2562ce1d0902.png)

So, if you looking closely the previous `rook-ceph-osd-prepare-` pods have changed their name in the last 4 characters. Hope this helps.